client/src/main/java/org/apache/falcon/client/FalconClient.java

```diff
@@ -1092,7 +1097,49 @@ private FormDataMultiPart getEntitiesForm(String extensionName, String jobName,
     private List<Entity> validateExtensionAndGetEntities(String extensionName, String jobName,
                                                          InputStream configStream) {
         ClientResponse clientResponse = getExtensionDetailResponse(extensionName);
-        List<Entity> entities = ExtensionHandler.getEntities(extensionName, jobName, configStream, clientResponse);
+        List<Entity> entities = getEntities(extensionName, jobName, configStream, clientResponse);
+        return entities;
+    }
+
+    private List<Entity> getEntities(String extensionName, String jobName, InputStream configStream,
+                                           ClientResponse clientResponse) {
+        JSONObject responseJson;
+        try {
+            responseJson = new JSONObject(clientResponse.getEntity(String.class));
+        } catch (JSONException e) {
+            OUT.get().print("Submit failed. Failed to get details for the given extension");
+            throw new FalconCLIException("Submit failed. Failed to get details for the given extension");
+        }
+        String extensionType;
+        String extensionBuildLocation;
+        try {
+            extensionType = responseJson.get("type").toString();
+            extensionBuildLocation = responseJson.get("location").toString();
+        } catch (JSONException e) {
+            OUT.get().print("Error. " + extensionName + " not found ");
+            throw new FalconCLIException("Submit failed. Failed to get details for the given extension");
+        }
+
+        List<Entity> entities = null;
+        if (!extensionType.equals(ExtensionType.CUSTOM.name())) {
+            try {
+                entities = ExtensionHandler.loadAndPrepare(extensionName, jobName, configStream,
+                        extensionBuildLocation);
+            } catch (Exception e) {
+                OUT.get().println("Error in building the extension");
+                throw new FalconCLIException("Submit failed. Failed to get details for the given extension");
+            }
+            if (entities == null || entities.isEmpty()) {
+                OUT.get().println("No entities got built");
+                throw new FalconCLIException("Submit failed. Failed to get details for the given extension");
+            }
+            try {
```

Comment:
The below part should move to ExtensionManager.submitInternal. Client shouldn't be doing that.